marlowe/src/Language/Marlowe/Core/V1/Merkle.hs

```diff
@@ -127,45 +126,66 @@ merkleize' f (When cases timeout contract) = When <$> mapM merkleizeCase cases <
 shallowDemerkleize
   :: (IsString e)
   => (MonadError e m)
-  => Contract
+  => Bool
+  -- ^ Throw an error if all required continuations are not present.
+  -> Contract
   -- ^ The contract.
   -> ReaderT Continuations m Contract
   -- ^ Action for the demerkleized contract.
-shallowDemerkleize = demerkleize' pure
+shallowDemerkleize = flip demerkleize pure
 
 -- | Demerkleize all case statements in a contract.
 deepDemerkleize
   :: (IsString e)
   => (MonadError e m)
-  => Contract
+  => Bool
+  -- ^ Throw an error if all required continuations are not present.
+  -> Contract
   -- ^ The contract.
   -> ReaderT Continuations m Contract
   -- ^ Action for the demerkleized contract.
-deepDemerkleize = fix demerkleize'
+deepDemerkleize = fix . demerkleize
 
 -- | Demerkleize selected case statements in a contract.
-demerkleize'
-  :: (IsString e)
+demerkleize
+  :: forall e m
+   . (IsString e)
   => (MonadError e m)
-  => (Contract -> ReaderT Continuations m Contract)
+  => Bool
```

Comment:
Cosmetics: That boolean alone is not really problematic (but harder to read on the call site). Let's say that when we have second `Bool` signature then we will introduce `newtype`s ;-)